mono/metadata/sgen-tarjan-bridge.c

```diff
@@ -115,7 +121,7 @@ typedef struct _ScanData {
 	// Tarjan algorithm index (order visited)
 	int index;
 	// Tarjan index of lowest-index object known reachable from here
-	int low_index : 27;
+	signed low_index : 27;
```

Comment:
Something horrible I learned while writing this patch: No, for bitfields, it isn't :( From the [C11 standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf), section 6.7.2 around page 111 specifies that "unsigned, or unsigned int" designate the same type and that "int, signed or signed int" also designate the same type:

> except that for bit-fields, it is implementation-defined whether the specifier int designates the same type as signed int or the same type as unsigned int.

This is reiterated several times in the document, and there is similar language in the C89 standard :(

It might be safe in practice but I'm considering going through and mass switching our `int` bitfields to explicitly signed.